internal/ui/input_glfw.go

```diff
@@ -78,7 +87,18 @@ var glfwMouseButtonToMouseButton = map[glfw.MouseButton]MouseButton{
 func (i *Input) update(window *glfw.Window, scale float64) {
 	i.m.Lock()
 	defer i.m.Unlock()
-
+	if i.runebuffer == nil {
+		i.runebuffer = make([]rune, 0, 1024)
+		window.SetCharModsCallback(func(w *glfw.Window, char rune, mods glfw.ModifierKey) {
+			if unicode.IsPrint(char) {
```

Comment:
Using IsPrint elides all spacing and control characters except ASCII space, so if you DebugPrint its output you don't get linefeed and tab glyphs rendered, so if that's not what you want (and it probably isn't) user code has to filter unprintables. You can make a case for arbitrary and typographical spacing if you're using a font that supports all graphical Unicode code points and rendering it in Ebiten, but particularly arbitrary spacing characters get hard to reason about and render in a useful way. There's no case to be made for legacy terminal control characters.